### PR TITLE
Add Bootstrap compiler interview

### LINKED ISSUE #73: Create a simple interview that compiles Bootstrap SCSS
Maybe it should also give you a template to customize the theme colors, but it should definitely let you upload a full SCSS file.

See: https://suffolklitlab.org/legal-tech-class/docs/practical-guide-docassemble/theming-docassemble#creating-a-custom-theme-from-source-instead-of-with-a-theme-generator

### PR DESCRIPTION
**Pull request by BryceStevenWilley**
Adds an interview that compiles bootstrap themes for admin and developers, according to https://suffolklitlab.org/legal-tech-class/docs/practical-guide-docassemble/theming-docassemble/#creating-a-custom-theme-from-source-instead-of-with-a-theme-generator.

Once per server, it installs bootstrap-5.2.3 in the `/tmp` dir, and then on going through the interview, it will run the `css-compile` command in the project, returning the standalone CSS file is there were no errors. The users can either upload a SCSS file, or type one directly, if it's shorter, or they want to iterate faster.

Also added to icon to the menu. There are better font-awesome icons (like a bootstrap specific one), but docassemble won't render brand ones correctly, so I'm sticking with a generic but unique `file-code` for now.

Fixes #73.

### CODE REVIEW

File: docassemble/ALDashboard/data/questions/compile_bootstrap.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  if not out_path.is_file():
+    error_screen
+  output_file.initialize(filename="custom.css")
+  output_file.copy_into(bootstrap_dir + f"dist/css/{file_name}.css")
+  compiled = True
+---
+id: start-page
+comment: |
+  Mostly here to prevent docassemble from auto downloading bootstrap in the playground / on load
+question: Create a custom theme for Bootstrap
```
Comment (nonprofittechy) [inline docassemble/ALDashboard/data/questions/compile_bootstrap.yml:72]:
```suggestion
question: |
  Create a custom theme for Bootstrap
```

File: docassemble/ALDashboard/data/questions/compile_bootstrap.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+subquestion: |
+  This interview will let you create a custom bootstrap theme, compiling it down into a single
+  CSS file that you can include in your docassemble projects.
+
+  For more information, see [our tutorial on making a custom bootstrap theme](https://suffolklitlab.org/legal-tech-class/docs/practical-guide-docassemble/theming-docassemble#creating-a-custom-theme-from-source-instead-of-with-a-theme-generator).
+continue button field: start_screen
+---
+id: file-upload
+question: What file do you want to make a theme from?
```
Comment (nonprofittechy) [inline docassemble/ALDashboard/data/questions/compile_bootstrap.yml:81]:
```suggestion
question: |
  What file do you want to make a theme from?
```

File: docassemble/ALDashboard/data/questions/compile_bootstrap.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+subquestion: |
+  Bootstrap returned this output when we tried to create your Bootstrap CSS file.
+
+  <pre>
+  ${ compile_output.stderr.decode() }
+  </pre>
+---
+event: done
+id: done
+question: Your file is compiled!
```
Comment (nonprofittechy) [inline docassemble/ALDashboard/data/questions/compile_bootstrap.yml:125]:
```suggestion
question: |
  Your file is compiled!
```

File: docassemble/ALDashboard/data/questions/compile_bootstrap.yml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      Type the contents of your bootstrap theme below
+    show if:
+      variable: upload_choice
+      is: type_out
+  - no label: file_text
+    input type: area
+    rows: 15
+    show if:
+      variable: upload_choice
+      is: type_out
```
Comment (nonprofittechy) [inline docassemble/ALDashboard/data/questions/compile_bootstrap.yml:107]:
```suggestion
      is: type_out
    default: |
        $white: #ffffff;
        $blue: #25dec6;
        
        $theme-colors: (
            "light":      #d8e2a5,
            "dark":       #1b1b1b,
            "primary":    #25dec6,
            "secondary":  #375b5a,
            "info":       #d74d72,
            "success":    #0cb545,
            "warning":    #f4ca0b,
            "danger":     #fa043c,
        );
        @import "bootstrap";
```

**Review by nonprofittechy — APPROVED** [review]
LGTM, just a couple low-hanging fruit suggestions you can accept or reject. Thanks for making this so quickly!

File: docassemble/ALDashboard/data/questions/menu.yml [hunk 5]
```diff
@@ -47,4 +47,6 @@ buttons:
   - Validate an attachment fields block: ${ interview_url(i=user_info().package + ":validate_attachment.yml", reset=1) }
     image: paperclip
   - PDF tools:  ${ interview_url(i=user_info().package + ":pdf_wrangling.yml", reset=1) }
-    image: file-pdf
\ No newline at end of file
+    image: file-pdf
+  - Bootstrap Theme Compiler: ${ interview_url(i=user_info().package + ":compile_bootstrap.yml", reset=1) }
```
Comment (nonprofittechy) [inline docassemble/ALDashboard/data/questions/menu.yml:51]:
```suggestion
  - Compile a custom Bootstrap theme: ${ interview_url(i=user_info().package + ":compile_bootstrap.yml", reset=1) }
```

**Comment by BryceStevenWilley** [comment]
FYI, the bootstrap logo issue will be fixed in https://github.com/jhpyle/docassemble/pull/614, which is why I didn't add an image or anything yet. This works though, we can take it out late maybe.

**Comment by BryceStevenWilley** [comment]
Minor note: the menu label for the bootstrap interview is a bit long. Doesn't matter much now, but if we add more or move it around, we'll have to shorten it or it'll be cut off.

Screenshot from 2023-01-31 19-25-50

**Comment by plocket** [comment]
> the menu label for the bootstrap interview is a bit long

I know the suggestion was actually to make it longer, but maybe something shorter would be sufficient? For example, some people may not be familiar with Bootstrap or compiling. `Custom CSS`? `Compile CSS`? `Custom theme`? And/or maybe we could reduce the padding in these buttons.

**Comment by BryceStevenWilley** [comment]
I agree with taking out Compiling, I would guess that more people would recognize Bootstrap over CSS though? Maybe just `Custom Bootstrap Theme`. Reducing padding would help a bit too.

**Pull request merged**